### PR TITLE
fix: SIGSEGV on artisan migrate when database connection is unavailable

### PR DESCRIPTION
**Pull request by goravel-coder**
## Summary
- Add `WithCommandsFilter` builder example to scope registered Artisan commands by environment
- Add integration tests verifying filter returns nil in local env and built-in commands survive
- Add production env test using config override + `Restart()` to verify filter activation

## Why
This adds an example demonstrating the `WithCommandsFilter` builder method introduced in goravel/framework#1500. The filter callback returns a positive list of command signatures to keep in production (using exact and glob matching), and returns nil to keep all commands in dev/local environments.

```go
func CommandsFilter() []string {
	if facades.Config().GetString("app.env") == "production" {
		return []string{
			"up",
			"down",
			"make:*",
			"vendor:publish",
		}
	}

	return nil
}
```

The filter is wired into the builder chain in `bootstrap/app.go`:

```go
return foundation.Setup().
	// ...
	WithCommands(Commands).
	WithCommandsFilter(CommandsFilter).
	// ...
```

### CODE REVIEW

File: database/migration/migrator.go [hunk 1]
```diff
@@ -33,6 +33,16 @@ func NewMigrator(artisan console.Artisan, schema contractsschema.Schema, table s
 	}
 }
 
+func (r *Migrator) requireOrm() error {
+	orm := r.schema.Orm()
+	if orm == nil || orm.Query() == nil {
+		color.Warningln(errors.SchemaOrmNotAvailable.Error())
```
Comment (hwbrzzl) [inline database/migration/migrator.go]: Is it necessary to print here given the error has been returned?

File: database/schema/schema.go [hunk 2]
```diff
@@ -413,6 +413,15 @@ func (r *Schema) Orm() contractsorm.Orm {
 	return r.orm
 }
 
+func (r *Schema) RequireOrm() error {
```
Comment (hwbrzzl) [inline database/schema/schema.go]: remove it if it's not used.

File: database/console/model_make_command.go [hunk 3]
```diff
@@ -72,6 +72,11 @@ func (r *ModelMakeCommand) Extend() command.Extend {
 }
 
 func (r *ModelMakeCommand) Handle(ctx console.Context) error {
+	if orm := r.schema.Orm(); orm == nil || orm.Query() == nil {
+		ctx.Error(errors.SchemaOrmNotAvailable.Error())
+		return nil
+	}
```
Comment (hwbrzzl) [inline database/console/model_make_command.go]: The code it duplicated, move it to a helper function.

**Review by hwbrzzl — COMMENTED** [review]
You are using `Maybe` in some tests, it's unexpected.
